src/language/dafnyInstallation.ts

```diff
@@ -89,15 +89,41 @@ export async function getCompilerRuntimePath(context: ExtensionContext): Promise
   return configuredPath;
 }
 
+// We cache the language server runtime path so that we don't need to copy it every time.
+let LanguageServerRuntimePath: string | null = null;
+
 export async function getLanguageServerRuntimePath(context: ExtensionContext): Promise<string> {
-  const configuredPath = await ifNullOrEmpty(
-    getConfiguredLanguageServerRuntimePath(),
+  if(LanguageServerRuntimePath != null) {
+    return LanguageServerRuntimePath;
+  }
+  const configuredLanguageServerPath = getConfiguredLanguageServerRuntimePath();
+  let mightBeRecomputed = false;
+  if(configuredLanguageServerPath !== '' && configuredLanguageServerPath != null) {
+    mightBeRecomputed = true;
+  }
+
+  let configuredPath = await ifNullOrEmpty(
+    configuredLanguageServerPath,
     async () => LanguageServerConstants.GetDefaultPath(await getConfiguredVersion(context))
   );
-  if(path.isAbsolute(configuredPath)) {
-    return configuredPath;
+  if(!path.isAbsolute(configuredPath)) {
+    configuredPath = path.join(context.extensionPath, configuredPath);
   }
-  return path.join(context.extensionPath, configuredPath);
+  if(mightBeRecomputed && (/\.dll$/.exec(configuredPath)) != null) {
```

Comment:
I suggest extracting the checking part into a function to make the intent clear to the reader.

For example:
```typescript
function isDll(path: string) {
  return /\.dll$/.test(configuredPath);
}
```

As you see, using `test(...)` simplifies the call since you don't have to check against `null`. That's possible since you only want to check if the pattern matches, not extracting the matching bits.

Last but not least, you may want to replace the regex entirely. Currently, you only want to check the suffix of the string, so you could use `endsWith` instead. In my opinion, that's more readable than a regex as long as it's a simple test like this:
For example:
```typescript
function isDll(path: string) {
  return configuredPath.endsWith('.dll');
}
```


PS: You may want to account for different cases. Then you may use your regex with `/i` to ignore the casing of the string.